pkg/p2p/peer/peer.go

```diff
@@ -329,10 +278,10 @@ func (p *Reader) ReadLoop(ctx context.Context) {
 	// 		log.Errorf("Peer %s failed with critical issue: %v", p.RemoteAddr(), r)
 	// 	}
 	// }()
-	p.readLoop(ctx)
+	p.readLoop(ctx, ringBuf)
 }
 
-func (p *Reader) readLoop(ctx context.Context) {
+func (p *Reader) readLoop(ctx context.Context, ringBuf *ring.Buffer) {
```

Comment:
using a private `readLoop` makes no sense here. Please incorporate this function in the exported one